zeus/monitor/energy.py

```diff
@@ -165,7 +184,7 @@ def __init__(
         #     1) Time elapsed at the beginning of this window.
         #     2) Total energy consumed by each >= Volta GPU at the beginning of
         #        this window (`None` for older GPUs).
-        self.measurement_states: dict[str, tuple[float, dict[int, float]]] = {}
+        self.measurement_states: dict[str, Measurement] = {}
```

Comment:
This is confusing code. I get that the `Measurement` class can actually hold all the state for a measurement window. However, when new contributors read this code, they will be seriously confused, because (1) `Measurement` is used both as measurement states and final measurements returned from `ZeusMonitor`'s public API, and (2) the `time` field, although documented as the *duration of the measurement window*, holds the beginning timestamp.

You should create a separate class called `MeasurementState`. This is a case when code duplication is actually justified. Plus, `MeasurementState` is not part of our public API anyway.

Reply:
Added a `MeasurementState` dataclass that is identical to the `Measurement` class but for keeping track of initial readings in `start_window`